### PR TITLE
Add rule BA2004

### PR DESCRIPTION
**Pull request by eddynaka**

### CODE REVIEW

File: src/BinSkim.Rules/PERules/WindowsBinaryAndPdbSkimmerBase.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics.SymbolStore;
```
Comment (michaelcfanning) [inline src/BinSkim.Rules/PERules/WindowsBinaryAndPdbSkimmerBase.cs:5]: >Diagnostics [](start = 13, length = 11)

I will remove this for now.

File: src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        {
+            // TODO: use DiaSymReader?
+        }
+
+        public void AnalyzeNativeBinaryAndPdb(BinaryAnalyzerContext context)
+        {
+            PEBinary target = context.PEBinary();
+            Pdb di = target.Pdb;
+
+            var compilandsWithOneOrMoreInsecureFileHashes = new List<ObjectModuleDetails>(); ;
```
Comment (eddynaka) [inline src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs:93]: >; ; [](start = 91, length = 3)

double ;

File: src/BinSkim.Driver/BinSkim.Driver.csproj [hunk 3]
```diff
@@ -26,7 +26,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="CommandLineParser" Version="2.8.0" />
+    <PackageReference Include="CommandLineParser" Version="2.7.82" />
```
Comment (eddynaka) [inline src/BinSkim.Driver/BinSkim.Driver.csproj:29]: >2.7.82 [](start = 59, length = 6)

do we need to downgrade?

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Comment by eddynaka** [comment]
Refs #319

**Pull request merged**